cmd/deploy.go

```diff
@@ -99,45 +99,37 @@ var deployCmd = &cobra.Command{
 		}
 
 		// create docker network or use existing one
-		if err = c.Runtime.CreateNet(ctx); err != nil {
+		if err = c.GlobalRuntime().CreateNet(ctx); err != nil {
 			return err
 		}
 
-		numNodes := uint(len(c.Nodes))
-		numLinks := uint(len(c.Links))
-		nodesMaxWorkers := maxWorkers
-		linksMaxWorkers := maxWorkers
+		nodeWorkers := uint(len(c.Nodes))
+		linkWorkers := uint(len(c.Links))
 
-		if maxWorkers == 0 {
-			nodesMaxWorkers = numNodes
-			linksMaxWorkers = numLinks
+		if maxWorkers > 0 && maxWorkers < nodeWorkers {
+			nodeWorkers = maxWorkers
 		}
 
-		if nodesMaxWorkers > numNodes {
-			nodesMaxWorkers = numNodes
-		}
-		if linksMaxWorkers > numLinks {
-			linksMaxWorkers = numLinks
+		if maxWorkers > 0 && maxWorkers < linkWorkers {
+			linkWorkers = maxWorkers
 		}
 
-		c.CreateNodes(ctx, nodesMaxWorkers)
-		c.CreateLinks(ctx, linksMaxWorkers, false)
-
-		// generate graph of the lab topology
-		if graph {
-			if err = c.GenerateGraph(topo); err != nil {
-				log.Error(err)
+		// Serializing ignite workers due to busy device error
+		for _, node := range c.Nodes {
+			if node.GetRuntime().GetName() == runtime.IgniteRuntime {
+				nodeWorkers = 1
+				break
```

Comment:
do we need a loop here? can we maybe have a look at `c.Runtimes` map and see if `runtime.IgniteRuntime` key is there?

Reply:
yep, you're right. will fix that